spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -64,8 +65,24 @@
  * <p>
  * This implementation will return the same {@link Producer} instance (if transactions are
  * not enabled) for the provided {@link Map} {@code configs} and optional {@link Serializer}
- * {@code keySerializer}, {@code valueSerializer} implementations on each
- * {@link #createProducer()} invocation.
+ * implementations on each {@link #createProducer()} invocation.
+ * <p>
+ * If you are using {@link Serializer}s that have no-arg constructors and require no setup, then simplest to
+ * specify {@link Serializer} classes in spring.kafka.producer configuration and create the
```

Comment:
Same here about wrong `spring.kafka.producer` reference where we need to mention appropriate Apache Kafka `ProducerConfig` props.